azure-pipelines.yml

```diff
@@ -137,28 +136,51 @@ steps:
   - task: Docker@2
     displayName: Login to temporary pipeline ACR
     inputs:
-      command: login
-      containerRegistry: $(PIPELINE_CONTAINER_REGISTRY)
+      containerRegistry: 'ASO Pipeline ACR'
```

Comment:
Just out of curiosity, it seems like there are a few different variables for this. There is still a "containerRegistry" variable set above. One task uses a variable called ["CONTAINER_REGISTRY_NAME"](https://github.com/Azure/azure-service-operator/blob/2dc02f8996c2ab3184bcb0097f0de1fb35ab3e57/azure-pipelines.yml#L291). Another task uses a variable called ["PIPELINE_ CONTAINER_REGISTRY_NAME"](https://github.com/Azure/azure-service-operator/blob/2dc02f8996c2ab3184bcb0097f0de1fb35ab3e57/azure-pipelines.yml#L230). Not saying that any of this is wrong, just that it's confusing which container registry is used and when. Do you know where the other variables come from and if they are they same as what you're setting here?

Reply:
I think some cleanup is due for sure. There are two container registries - one is a temporary pipeline registry that we push the image to and use for testing on the cluster, and only if the deploy test succeeds we push the image to a release ACR, which is the other one. Happy to look at consolidating some of these

Reply:
Addressed this and also added the variables needed to be configured at the begining of the pipeline for clarity